testng-core/src/main/java/org/testng/internal/invokers/TestInvoker.java

```diff
@@ -848,50 +848,58 @@ public ITestResult registerSkippedTestResult(
     return result;
   }
 
-  private static final Object LOCK = new Object();
+  private static final AutoCloseableLock LOCK = new AutoCloseableLock();
```

Comment:
Google uses UPPER CASE for constants only, see https://google.github.io/styleguide/javaguide.html#s5.2.4-constant-names

`AutoCloseableLock` instance is mutable, thus even `static` field should better be named as `lock`. Of course, it is up to you, however, I find Google suggestion nice and documented.